src/Export/ProductService.php

```diff
@@ -26,33 +28,45 @@ class ProductService
 {
     public const CONTAINER_ID = 'fin_search.product_service';
 
+    /** @var Config|null */
+    private $config;
+
     /** @var ContainerInterface */
     private $container;
 
     /** @var SalesChannelContext|null */
     private $salesChannelContext;
 
-    public function __construct(ContainerInterface $container, ?SalesChannelContext $salesChannelContext = null)
-    {
+    public function __construct(
+        ContainerInterface $container,
+        ?SalesChannelContext $salesChannelContext = null,
+        ?Config $config = null
+    ) {
         $this->container = $container;
         $this->salesChannelContext = $salesChannelContext;
+        $this->config = $config ?? $container->get(Config::class);
     }
 
     public static function getInstance(
         ContainerInterface $container,
-        ?SalesChannelContext $salesChannelContext
+        ?SalesChannelContext $salesChannelContext,
+        ?Config $config = null
     ): ProductService {
         if ($container->has(self::CONTAINER_ID)) {
             $productService = $container->get(self::CONTAINER_ID);
         } else {
-            $productService = new ProductService($container, $salesChannelContext);
+            $productService = new ProductService($container, $salesChannelContext, $config);
             $container->set(self::CONTAINER_ID, $productService);
         }
 
         if ($salesChannelContext && !$productService->getSalesChannelContext()) {
             $productService->setSalesChannelContext($salesChannelContext);
         }
 
+        if ($config && !$productService->getConfig()) {
```

Comment:
When would this be the case, shouldn't the constructor already set this? Would we even need a `setConfig` and `getConfig` method?

Reply:
yes it does set this in a real live scenario, but this is following the sales channel pattern. It especially is helpful in tests where we do not have to create the whole instance of the service again and can just work with a mocked config. This if block is most probably unnecessary yes, since this would never be null eventually, but the `get/set` methods would be needed still.